modules/3d/src/pointcloud/load_point_cloud.cpp

```diff
@@ -0,0 +1,108 @@
+#include "../precomp.hpp"
+
+#include "coders_base.hpp"
+#include "coders_obj.hpp"
+#include "coders_ply.hpp"
+
+#include "opencv2/core/utils/filesystem.hpp"
+#include "opencv2/core/utils/filesystem.private.hpp"
+
+#include <memory>
+#include <iostream>
+
+namespace cv
+{
+
+namespace pc
+{
+
+#if OPENCV_HAVE_FILESYSTEM_SUPPORT
+
+static PointCloudDecoder findDecoder(const String &filename)
+{
+    size_t filename_length = filename.length();
+    if ((filename.substr(filename_length - 4, 4) == ".obj") || (filename.substr(filename_length - 4, 4) == ".OBJ"))
+    {
+        return std::unique_ptr<ObjDecoder>(new ObjDecoder());
+    }
+    if ((filename.substr(filename_length - 4, 4) == ".ply") || (filename.substr(filename_length - 4, 4) == ".PLY"))
+    {
+        return std::unique_ptr<PlyDecoder>(new PlyDecoder());
+    }
+
+    return nullptr;
+}
+
+static PointCloudEncoder findEncoder(const String &filename)
+{
+    size_t filename_length = filename.length();
+    if (filename.substr(filename_length - 4, 4) == ".obj")
+    {
+        return std::unique_ptr<ObjEncoder>(new ObjEncoder());
+    }
+    if (filename.substr(filename_length - 4, 4) == ".ply")
+    {
+        return std::unique_ptr<PlyEncoder>(new PlyEncoder());
+    }
+
+    return nullptr;
+}
+
+#endif
+
+}
+
+void loadPointCloud(const String &filename, OutputArray vertices, OutputArray normals)
+{
+#if OPENCV_HAVE_FILESYSTEM_SUPPORT
+    pc::PointCloudDecoder decoder;
+
+    decoder = pc::findDecoder(filename);
+
+    decoder->setSource(filename);
+
+    std::vector<Point3f> vec_vertices;
+    std::vector<Point3f> vec_normals;
+
+    decoder->readData(vec_vertices, vec_normals);
```

Comment:
`readData` throws exceptions with `CV_Error` macro. It means that function just falis each time when file was not open/empty/etc. It's against your function description (fail-safe execution and empty arrays as output).